shared/lib-hoppy/hoppy/base_channel_opener.py

```diff
@@ -78,8 +78,7 @@ def _reconnect(self):
         self.stop()
         reconnect_delay = self._get_reconnect_delay()
         self._warning('reconnecting', reconnect_delay_seconds=reconnect_delay)
-        time.sleep(reconnect_delay)
-        self.connect(self._custom_loop)
+        self._custom_loop.call_later(reconnect_delay, self.connect, self._custom_loop)
```

Comment:
Nice work!

Question (non-blocking): normally we use `asyncio.sleep` for this kind of case - is there a reason you prefer `call_later` here, and should we look at changing any of the other sleep calls to use `call_later` as well?

Reply:
using this call later method means I don't have to await the asyncio.sleep call!